tests/test_core.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+import pytest
+
+from mewbot.core import (
+    ComponentKind,
+    BehaviourInterface,
+    TriggerInterface,
+    ConditionInterface,
+    ActionInterface,
+    IOConfigInterface,
+)
+
+# pragma pylint: disable=R0903
+#  Disable "too few public methods" for test cases - most test files will be classes used for
+#  grouping and then individual tests alongside these
+# pragma pylint: disable=R0201
+#  Disable "no self use" for functions. These functions will not be used internally as they are
+#  automatically called by pytest as it seeks and searches for tests.
```

Comment:
You can avoid having to do this be removing the class. As far as I can tell, having the tests in a class doesn't do anything in terms of the information pytest returns.

Reply:
(Regarding fixing, it probably ends up as a convention thing; I personally would group tests into classes, particularly because that allows commonality with some configuration settings for some cases, and then the other cases match the same convention - but that's a Kitsune stylistic thing.)

The class containing the tests is written into output messages when the tests fail.

For example, if I simulate an error in the function `test_componentkind_interface_map_ioconfig`, it says: `TestComponent.test_componentkind_interface_map_ioconfig` and `FAILED tests/test_core.py::TestComponent::test_componentkind_interface_map_ioconfig - AssertionError:...`

rather than removing `test_componentkind_interface_map_ioconfig` from it's class, instead I get: `test_componentkind_interface_map_ioconfig` and `FAILED tests/test_core.py::test_componentkind_interface_map_ioconfig - AssertionError:...`

Which only really matters for visually seeing where errors are, rather than the automated output system in `tools/test.py`

Reply:
I find grouping tests into classes helpful in a number of ways
 - It allows for (easier) automated test generation. This can be particularly useful when trying to optimize (later - "premature optimization" and all) when, for example, you have a newly written optimized component which you want to test with a bunch of other configs to make sure everything all works
 - Given this program's going to be all config driven, I suspect being able to subclass tests for slightly different configs will be really helpful
 - It allows (again, possibly just me personally) to derive useful stuff from them more easily (e.g. "this started life as a test suite. Now it's part of my optimization and profiling toolchain")

So yes! I am opinionated in favour of test classes.